33 address running tests and laravel start time

dstepe opened the pull request:
This is a draft PR so we can discuss this proposed change.

I am concerned about the use of the `LARAVEL_START` constant. Not only does this cause problems in feature test scenarios, but that constant does not appear to be explicitly exposed by the framework in a sustainable way. Additional, this module reimplements timing and duration calculations for use in the APM transaction which could be provided with an existing Timer class.

This change does require an update to the elastic-apm-php-agent dependency.

The updated Timer class can accept an explicit start time. The PHP provided `REQUEST_TIME_FLOAT` appears to be the appropriate source for that start time. The popular Laravel DebugBar plugin has stopped using LARAVEL_START in favor of this approach.

The service provider will register a new Timer object and ensure it is correctly initialized. The service provider will also set a private start time member to the same value for use in query timing.

I believe this change is consistent with the Laravel framework conventions and consolidates APM related timing calculations.

tepeds committed:
Use REQUEST_TIME_FLOAT to start transaction

tepeds committed:
Provide Timer object through service container

tepeds committed:
Remove unused getDuration method; add notes about testing to README

philkra commented:
@dstepe  would you be interested to become a collabotrator for this repo?

dstepe commented:
Yes, I'm open to that. We have a number of Laravel applications and are just getting started with APM, so I'm highly motivated right now. This module really helped me get started quickly and I'm happy to help work on it.

philkra commented:
The baseline PR of the agent is merged.

philkra commented on composer.json:
```diff
@@ -16,7 +16,7 @@
       "illuminate/routing":"5.5.x|5.6.x|5.7.x",
       "illuminate/support":"5.5.x|5.6.x|5.7.x",
-      "philkra/elastic-apm-php-agent":"6.*.*",
+      "philkra/elastic-apm-php-agent":"dev-48-allow-explicit-timer-start",
```
I know this is just for testing, but please add min version `6.5.3` of the agent, to ensure the client is actually able to perform what it should.

philkra reviewed (changes requested):
see review comment

tepeds committed:
Update to new required release of elastic-apm-php-agent

The new release supports the Timer class using a provided start time which we need to use the REQUEST_TIME_FLOAT value

tepeds committed:
Remove dependency vcs repo from composer.json

dstepe commented:
The composer file is cleaned up and ready.

dstepe commented:
We've been running this in a Laravel app here in test for a few days and have not seen any problems. What's your normal process for testing and validation before merging and releasing?

philkra commented on composer.json:
```diff
@@ -16,7 +16,7 @@
       "illuminate/routing":"5.5.x|5.6.x|5.7.x",
       "illuminate/support":"5.5.x|5.6.x|5.7.x",
-      "philkra/elastic-apm-php-agent":"dev-48-allow-explicit-timer-start",
+      "philkra/elastic-apm-php-agent":">=6.5.3",
```
perfect!

philkra commented:
Thanks for the contribution @dstepe and in case you are doing another PR for this repo, please add yourself to the contributors in the `composer.json` file.
